src/EditorFeatures/CSharpTest/RemoveUnusedVariable/RemoveUnusedVariableTests.cs

```diff
@@ -308,5 +308,276 @@ bool TrySomething()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    int M()
+    {
+        int [|unused|] = 0;
+        int used = 0;
+        unused = used = 0;
+        return used;
+    }
+}
+",
+@"
+class C
+{
+    int M()
+    {
+        int used = 0;
+        used = 0;
+        return used;
+    }
+}
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedVariable)]
+        public async Task JointAssignmentRemoveSecond()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    int M()
+    {
+        int used = 0;
+        int [|unused|] = 0;
+        used = unused = 0;
+        return used;
+    }
+}
+",
+@"
+class C
+{
+    int M()
+    {
+        int used = 0;
+        used = 0;
+        return used;
+    }
+}
+");
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/22921"), Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedVariable)]
+        public async Task RemoveUnusedLambda()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    int M()
+    {
+        Func<int> [|unused|] = () =>
+        {
+            return 0;
+        };
+        return 1;
+    }
+}
+",
+@"
+class C
+{
+    int M()
+    {
+        return 1;
+    }
+}
+");
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.CodeActionsSimplifyTypeNames)]
+        [Trait(Traits.Feature, Traits.Features.CodeActionsFixAllOccurrences)]
+        public async Task JointDeclarationRemoveBoth()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>
+class C
+{
+    int M()
+    {
+        int {|FixAllInDocument:a|} = 0, b = 0;
+        return 0;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>
+";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>
+class C
+{
+    int M()
+    {
+        return 0;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>
+";
+
+            await TestInRegularAndScriptAsync(input, expected);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.CodeActionsSimplifyTypeNames)]
+        [Trait(Traits.Feature, Traits.Features.CodeActionsFixAllOccurrences)]
+        public async Task JointAssignmentRemoveBoth()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>
+class C
+{
+    int M()
+    {
+        int a = 0;
+        int {|FixAllInDocument:b|} = 0;
+        a = b = 0;
+        return 0;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>
+";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>
+class C
+{
+    int M()
+    {
+        int a = 0;
```

Comment:
nit: I believe the behavior of FixAll is correct here as the compiler does not mark local 'a' as unused here for `a = b = 0`. Do we want to rename the test to not end with `RemoveBoth`?